src/main/java/uk/gov/companieshouse/api/interceptor/CRUDAuthenticationInterceptor.java

```diff
@@ -36,15 +37,19 @@
     boolean enableTokenPermissionAuth;
 
     private final Permission.Key permissionKey;
+    private final boolean ignoreAPIKeyRequests;
     private final Set<String> ignoredHttpMethods;
 
     /**
      * 
      * @param permissionKey The expected permission key
+     * @param ignoreAPIKeyRequests If true this interceptor will allow any API key traffic through.
+     *         Other specific API key checks (for elevated privileges etc) should be applied to these routes to cover specific logic when this is true.
      * @param ignoredHttpMethods An optional array of http methods for which the interceptor won't run
      */
-    public CRUDAuthenticationInterceptor(Permission.Key permissionKey, String... ignoredHttpMethods) {
+    public CRUDAuthenticationInterceptor(Permission.Key permissionKey, boolean ignoreAPIKeyRequests, String... ignoredHttpMethods) {
```

Comment:
I would overload this constructor so that it won't break the clients when we upgrade the library version?

```
public CRUDAuthenticationInterceptor(Permission.Key permissionKey, String... ignoredHttpMethods) {
    this(permissionKey, false, ignoredHttpMethods);
}
````